### DISCUSSION TITLE
Mosmix Standort Problem - Anfänger

### DISCUSSION THREAD

**Discussion by timi35** [roe-dl/weewx-DWD#54] [Q&A]
Hallo,
ich nutze folgende Erstkonfig.
Starte ich nun dwd-mosmix --weewx, bekomme ich immer Döbeln als Forecast.
Was übersehe ich bzw. mache ich falsch?

Vielen Dank!
Tim

[DeutscherWetterdienst]
    # Konfiguration für dwd-cap-warnings
    [[warning]]
        icons='../dwd/warn_icons_50x50'
        states='Sachsen'
        resolution = county|city
        [[[counties]]]
              'Stadt Dresden'='DD'
        [[[cities]]]
              'Dresden-Altstadt'='DD'
[WeatherServices]
    # Verzeichnis, in das die Dateien gespeichert werden sollen
    path='/etc/weewx/skins/Belchertown/dwd'
    # Konfiguration zum Download aktueller Meßwerte von offiziellen
    # Stationen
    [[current]]
        [[[10488]]]
            provider = DWD
            model = POI
    # Konfiguration für dwd-mosmix
    [[forecast]]
        # Speicherort der Symbole auf dem Web-Server
        icons = '../images'
        # zu verwendender Symbolsatz: belchertown, dwd, aeris
        icon_set = belchertown
        # für welche Orientierungen soll die HTML-Datei erzeugt werden?
        orientation = h,v
        # Sollen die Meßgrößensymbole in die Tabelle aufgenommen werden?
        show_obs_symbols = True # optional
        # Soll die Meßgrößenbeschreibung in die Tabelle aufgenommen werden?
        show_obs_description = False # optional
        # Soll der Ortsname über die Tabelle geschrieben werden?
        show_placemark = True # optional
    # Warnungen (dwd-cap-warnings, bbk-warnings, msc-warnings)
    [[warning]]
        #icons = ... # Optional, für alle Provider
        #bbk_icons = ... # Optional, nur für BBK
        #bbk_logos = ... # Optional, nur für BBK
        #bbk_include_dwd ... # Optional, nur für BBK
        #dwd_icons = ... # Optional, nur für DWD
        #dwd_status_url = ... # Optional, nur für DWD
        #dwd_diff_url = ... # Optional, nur für DWD
     [[[Meissen]]]
            provider = BBK
            county = 146270000000
            file = MEI
     # Konfiguration für die --belchertown Option von dwd-mosmix
     [[Belchertown]]
        # Name des Abschnittes der Belchertown-Skin in [StdReport]
        section = Belchertown
        # Warndatei aus dem Abschnitt [[warnings]]
        warnings = MEI
        # Vorhersagedatei aus dem Aufruf 'dwd-mosmix'
        forecast = K3384
        # schließe Warnungen ein, die die angegebene Zahl von Sekunden in
        # der Zukunft gültig werden (optional)
        #include_advance_warnings = 0 
        # Quelle für Luftqualtitätsdaten (optional)
        # mögliche Werte: aeris, ubaXXXX
        #aqi_source = ...
        # Sprache für die Himmelsrichtungen (optional)
        # mögliche Werte: de, en, fr, it, cz, es, nl, no, gr
        #compass_lang = 'de' # optional

**Comment by roe-dl** [comment]
Beim Aufruf von `dwd-mosmix` muß der Code des Ortes angegeben werden, für den die Vorhersage berechnet werden soll. Die Codes sind im [Wiki](https://github.com/roe-dl/weewx-DWD/wiki/MOSMIX-Stationen-in-Deutschland-(Ost)) beschrieben, für Dresden u.a. `10487`

**Comment by timi35** [comment]
Vielen  Dank, dann war das ein Denkfehler.
ich war davon ausgegangen,wenn ich --weewx mitgebe, dass dann automatisch 

        # Vorhersagedatei aus dem Aufruf 'dwd-mosmix'
        forecast = 10488

ausgelesen  und verwendet.

Ich habe jetzt testweise den stündlochen Crontab drin und habe die manuell Variante gewählt um die Vorhersage und Warnungen einzubinden.
Wie bekomme ich aber die belchertown 8 Tage Wettervorschau zum Laufen?
Das hat offensichtlich 1x funktioniert, jedoch steht sie seitdem auf dem alten Stand.

Momentan sieht es so aus?

        section = Belchertown
        # Warndatei aus dem Abschnitt [[warnings]]
        warnings = MEI   [[Belchertown]]
  
        # Vorhersagedatei aus dem Aufruf 'dwd-mosmix'
        forecast = 10488
        # schließe Warnungen ein, die die angegebene Zahl von Sekunden in
        # der Zukunft gültig werden (optional)
        #include_advance_warnings = 0 
        # Quelle für Luftqualtitätsdaten (optional)
        # mögliche Werte: aeris, ubaXXXX
        #aqi_source = ...
        # Sprache für die Himmelsrichtungen (optional)
        # mögliche Werte: de, en, fr, it, cz, es, nl, no, gr
        #compass_lang = 'de' # optional

und unter Extras:


            # Forecast defaults
            forecast_enabled = 1
            #forecast_provider = "darksky"
            #forecast_api_id = "xxxxxxx_deleted"
            #forecast_api_secret = "xxxxxxx_deleted"
            forecast_units = "ca"
            forecast_lang = "de"
            forecast_stale = 86400
            forecast_alert_enabled = 1
            forecast_alert_limit = 1
            forecast_show_daily_forecast_link = 0
            forecast_daily_forecast_link = ""


Wodurch wird die Aktualiserung der 8 Tage Vorrauschau "angeschoben"?

Vielen Dank!
VG Tim

**Comment by roe-dl** [comment]
Ich bin mir nicht sicher, ob `forecast_provider` auf irgendeinen Wert gesetzt werden muß. Was genau wird denn zur Zeit angezeigt? Und mit welchen Parametern wird `dwd-mosmix` regelmäßig aufgerufen?

**Comment by timi35** [comment]
Hallo,

im hourly cron läuft 

/usr/local/bin/dwd-mosmix --weewx --daily --hourly 10488 2>/dev/null >/dev/null

Was für die manuelle Vorschau per index_hook_after_forecast.inc auch gut funktioniert. Wenn ich aber die 8 Tage Vorschau nutzen will, bleibt es beim 18.7. 19:47 stehen.
Es hat also einmal die Daten gezogen, aktualisiert aber nicht.

Vielen Dank!
VG Tim

**Comment by timi35** [comment]
Ich glaube ich kann es eingrenzen.

Nutze ich:
/usr/local/bin/dwd-mosmix --weewx --daily --hourly 10488 2>/dev/null >/dev/null

erhalte ich: forecast-10488-hourly.inc, forecast-10488.json und forecast-10488.inc

Nutze ich:
sudo /usr/local/bin/dwd-mosmix --weewx --belchertown 10488 2>/dev/null >/dev/null
erhalte ich nur die geo-10488.json

Nutze ich:

/usr/local/bin/dwd-mosmix --weewx --daily --hourly 10487 2>/dev/null >/dev/null

erhalte ich: forecast-10487-hourly.inc, forecast-10487.json und forecast-10487.inc

Bei: /usr/local/bin/dwd-mosmix --weewx --belchertown 10487 2>/dev/null >/dev/null

erhalte ich die wohl für die 8 Tage Vorschau benötigte forecast-10487-belchertown.json.

Es ist also stationsabhängig, das ist sehr seltsam.
Ich habe jetzt testweise in stündl. Crontab

/usr/local/bin/dwd-mosmix --weewx --daily --hourly 10487 2>/dev/null >/dev/null
/usr/local/bin/dwd-mosmix --weewx --belchertown 10487 2>/dev/null >/dev/null

VG Tim

**Reply by roe-dl** [reply]
Das Ergebnis der Option `--belchertown` heißt `forecast.json` und wird dorthin geschrieben, wo es von der Belchertown-Skin erwartet wird, nämlich im Unterverzeichnis `json` (typischerweise `/var/www/html/belchertown/json`). Das ist nur für genau eine Station möglich, denn die Skin kann nur die Daten einer Station anzeigen. Es ist die Station, die im Abschnitt `[[Belchertown]]` im Schlüssel `forecast` angegeben ist. Für alle anderen Stationen wird hilfsweise eine Datei `forecast-Station-belchertown.json` erzeugt. Die ist aber nutzlos, weil sie nicht weiter verarbeitet werden kann.

Pro Station ist nur ein einziger Aufruf von `dwd-mosmix` nötig. Die Optionen `--daily`, `--hourly` und `--belchertown` können kombiniert werden.

**Comment by timi35** [comment]
Hallo,

habe es inzwischen hinbekommen. Wenn man die eine index_hook nutzt, muss man natürlich auch dort drin ändern.
Momentan sieht also wie folgt aus und funktioniert.
Rufe ich jedoch dwd-mosmix im cron kombiniert auf, wird eine der beiden Wetterprognosen nicht aktualisiert.
Getrennt schon, was okay für mich ist.

Datei index_hook_after_forecast.inc


  ```
<div class="col-sm-8">
    <p style="font-size:110%">Wettervorhersage</p>
    #include raw "dwd/forecast-10487.inc"
  </div>

  <div class="col-sm-12" style="margin-bottom:1em">
    #if os.path.exists("dwd/VHDL50_DWLG_LATEST.html")
    #include raw "dwd/VHDL50_DWLG_LATEST.html"
    [Quelle:
    <a
    href="https://www.dwd.de/DE/wetter/wetterundklima_vorort/sachsen/sac_node.h>
    target="_blank">DWD</a>]
    #end if
  </div>


  <div class="col-sm-6">
    <p style="font-size:110%">Wetterwarnungen</p>
    #include raw "dwd/warn-MEI.inc"
  </div>

```

/etc/cron.hourly:

```
#!/bin/bash
# Daten beim Deutschen Wetterdienst herunterladen
# Copyright (C) 2021, 2023 Johanna Roedenbeck
# licensed under the terms of the General Public Licens (GPL) v3

# Diese Datei ist unter /etc/cron.hourly zu speichern. Sie muss
# ausfuehrbar sein.

/usr/local/bin/wget-dwd 2>/dev/null
/usr/local/bin/dwd-cap-warnings --weewx --resolution=county 2>/dev/null >/dev/null

# Soll statt dwd-cap-warnings dwd-warnings verwendet werden,
# hier das Kommentarzeichen entfernen und dafuer eines vor 
# die Zeile mit dwd-cap-warnings setzen
#/usr/local/bin/dwd-warnings 2>/dev/null >/dev/null

# Fuer Wettervorhersagen XXXXX durch den Stationscode
# ersetzen.
/usr/local/bin/dwd-mosmix --weewx --daily --hourly 10487 2>/dev/null >/dev/null
/usr/local/bin/dwd-mosmix --weewx --belchertown 10487 2>/dev/null >/dev/null

exit 0
```

weewx.conf:

```
**[DeutscherWetterdienst]
    # Konfiguration für dwd-cap-warnings
    [[warning]]
        icons='../belchertown/dwd-icons/warn_icons_50x50'
        states='Sachsen'
        resolution = county|city
        [[[counties]]]
              'Kreis Meißen'='MEI'
        [[[cities]]]
              'Dresden-Altstadt'='DD'
[WeatherServices]
    # Verzeichnis, in das die Dateien gespeichert werden sollen
    path='/etc/weewx/skins/Belchertown/dwd'
    # Konfiguration zum Download aktueller Meßwerte von offiziellen
    # Stationen
 
     #[[current]]
     #       [[[10488]]]
     #       provider = DWD
     #       model = POI
  
 
    # Konfiguration für dwd-mosmix
    [[forecast]]
        # Speicherort der Symbole auf dem Web-Server
        icons = '../images'
        # zu verwendender Symbolsatz: belchertown, dwd, aeris
        icon_set = belchertown
        # für welche Orientierungen soll die HTML-Datei erzeugt werden?
        orientation = h,v
        # Sollen die Meßgrößensymbole in die Tabelle aufgenommen werden?
        show_obs_symbols = True # optional
        # Soll die Meßgrößenbeschreibung in die Tabelle aufgenommen werden?
        show_obs_description = False # optional
        # Soll der Ortsname über die Tabelle geschrieben werden?
        show_placemark = False # optional
    # Warnungen (dwd-cap-warnings, bbk-warnings, msc-warnings)
    [[warning]]
        #icons = ... # Optional, für alle Provider
        #bbk_icons = ... # Optional, nur für BBK
        #bbk_logos = ... # Optional, nur für BBK
        #bbk_include_dwd ... # Optional, nur für BBK
        #dwd_icons = ... # Optional, nur für DWD
        #dwd_status_url = ... # Optional, nur für DWD
        #dwd_diff_url = ... # Optional, nur für DWD

        [[[Meissen]]]
            provider = BBK
            county = 146270000000
            file = MEI
            
            
     [[Belchertown]]           
     # Konfiguration für die --belchertown Option von dwd-mosmix
        # Name des Abschnittes der Belchertown-Skin in [StdReport]
        section = "Belchertown"
        # Warndatei aus dem Abschnitt [[warnings]]
        warnings = MEI   
    # Vorhersagedatei aus dem Aufruf 'dwd-mosmix'
    forecast = 10487
        # schließe Warnungen ein, die die angegebene Zahl von Sekunden in
        # der Zukunft gültig werden (optional)
        #include_advance_warnings = 0 
        # Quelle für Luftqualtitätsdaten (optional)
        # mögliche Werte: aeris, ubaXXXX
        #aqi_source = ...
        # Sprache für die Himmelsrichtungen (optional)
        # mögliche Werte: de, en, fr, it, cz, es, nl, no, gr
        #compass_lang = 'de' # optional**


```
